website/docs/HowShouldIQuery.md

```diff
@@ -3,19 +3,143 @@ id: how-should-i-query
+
+In most cases, you need to set accessibility roles explicitly (or your component library can set some of them for you). These roles allow assistive technologies (like screen readers) and testing code to understand your view hierarchy better.
+
+Some frequently used roles include:
+
+- `alert` - important text to be presented to the user, e.g., error message
+- `button`
+- `checkbox` & `switch` - on/off controls
+- `heading` (`header`) - header for content section, e.g., the title of navigation bar
+- `img` (`image`)
+- `link`
+- `menu` & `menuitem`
+- `progressbar`
+- `radiogroup` & `radio`
+- `searchbox` (`search`)
+- `slider` (`adjustable`)
+- `summary`
+- `tablist` & `tab`
+- `text` - static text that cannot change
+- `toolbar` - container for action buttons
+
+#### Name option {#by-role-query-name-option}
+
+Frequently, you will want to add the [`name`](api-queries#by-role-options) option, which will match both the element's role and its accessible name (= element's accessibility label or text content).
+
+Here are a couple of examples:
+
+- button Start : `getByRole("button", { name: "Start" })`
+- silent mode switch: `getByRole("switch", { name: "Silent Mode" })`
+- whole menu: `getByRole("menu", { name: "Editing Menu" })`
+- particular menu item: `getByRole("menuitem", { name: "Undo" })`
+- error messages: `getByRole("alert", { name: /Not logged in/ })`
+- screen header: `getByRole("header", { name: "Settings" })`
+
+#### Other options {#by-role-query-other-options}
+
+Other useful [options](api-queries#by-role-options) include:
+
+- accessibility states like: `disabled`, `selected`, `checked`, `busy`, `expanded`
+- accessibility value: `value: { now, min, max, text }`
+
+These option types can frequently be better expressed by using the corresponding Jest matchers:
+
+- [`toBeEnabled()` / `toBeDisabled()`](jest-matchers#tobeenabled)
+- [`toBeSelected()`](jest-matchers#tobeselected)
+- [`toBeChecked()` / `toBePartiallyChecked()`](jest-matchers#tobechecked)
+- [`toBeExpanded()` / `toBeCollapsed()`](jest-matchers#tobeexpanded)
+- [`toBeBusy()`](jest-matchers#tobebusy)
+- [`toHaveAccessibilityValue()`](jest-matchers#tohaveaccessibilityvalue)
+
+### 2. Text input queries {#text-input-queries}
+
+Querying [`TextInput`](https://reactnative.dev/docs/textinput) elements presents a unique challenge as there is no separate role for `TextInput` elements. There is a `searchbox`/`search` role, which can be assigned to `TextInput`, but it should be only used in the context of search inputs, leaving other text inputs without a role to query with.
+
+Therefore, you can use the following queries to find relevant text inputs:
+
+1. [`*ByLabelText`](api-queries#by-label-text) - will match the accessibility label of the element. This query will match any host elements, including `TextInput` elements.
```

Comment:
Is it a good practice to use an a11y label on a text input? You'll need to change it depending on the value it's not very practical. I personally don't and use `ByDisplayValue ` and `ByPlaceholderText` almost all of the time. I also like those more because they're specific to `TextInput` so they make tests clearer imo

Reply:
Good question. I think that every TextInput needs to have a label identifying its purpose. For visual users it might be a adjacent `Text` label, however for users using a11y this might not so evident, and hence setting a11y label would benefit them, as well as improve the testing. 
(Note to self: I need to verify whether `placeholder` prop is pronounced under screen reader.)

Regarding `*ByPlaceholder` vs `*ByDisplayValue`, I would prefer using placeholder query as it identifies the input by it's function, regardless of current value which might change.